# CCS-Metatranscriptome

#### Metatranscriptome bioinformatic pipeline for eukaryotic phytoplankton from the California Current System.

by Yuan Yu (Johnson) Lin

## Introduction - Diatoms exhibit proactive transcriptomic response to the Upwelling Conveyor Belt Cycle (UCBC)

Phytoplankton in the California Current System (CCS) are important drivers of biogeochemical cycling and food web dynamics leading up to fish stocks and marine mammals - important aspects of the ecosystem and economy. A unique feature of the CCS is the concept of the Upwelling Conveyor Belt Cycle, which is characterized by a series of idealized zones directly related to the respective light conditions, nutrient status, and biology. Diatoms - a silicifying taxa of the CCS phytoplankton assemblage - are known to dominante the upwelling blooms in this region due to their physiological capabilities. However, our understanding of the molecular shifts within the major phytoplankton groups remain largely understudied. This study aims to address the bioinformatic tools used to assess the transcriptomic and metatranscriptomic activity of marine microalgae in response to the UCBC.

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

## Bioinformatic Pipeline

### Introduction

Sequencing is a burgeoning tool used to characterize marine microorganisms in the world's oceans. As a result, phytoplankton groups are still poorly annotated given the relatively scarce reference genomes/transcriptomes out there. However, as RNA-seq is increasingly used in oceanography, our knowledge of 'omics-based tools will continue to expand. In our current state, we can still derive a great source of knowledge from the large datasets collected from the ocean environment. 

Below is a detailed breakdown of the workflow used to evaluate the metatranscriptomics of the coastal phytoplankton communities. Prior to this workflow, you will have obtained the raw reads (sample.fastq.gz) generated by Illumina Hiseq. This pipeline will be run exclusively on the Linux-based Longleaf cluster at the University of North Carolina, Chapel Hill.

### 1. Quality Control

The first step of the workflow will be to get rid of bad reads and adapter sequences by trimming. There is an abundance of adapter trimming tools out there, but for the purpose of our study, we will use Trim Galore.


```
#!/bin/bash

indir=/proj/marchlab/projects/PUPCYCLE_2019/PUP_SimUp/raw/00_fastq
outdir=/proj/marchlab/projects/PUPCYCLE_2019/PUP_SimUp/galore_test

samples='J1 J2 J3 J4 J5 J6 J7 J8 J9 J10 J11 J12 J13 J14 J15 J16 J17 J18 J20 J21 J22 J23 J24 J25 J26 J27 J28 J29 J30 J31 J40 J41 J42 J43 J44 J45 J46 J47 J48 J49 J50 J51 J52 J53 J54 J55 J56 J57 J60 J61 J62 J63 J64 J65 J66 J67 J68 J69 J70 J71 J72 J73 J74 J75 J76 J77'

for s in $samples; do
        echo ${s}
	jobfile="trim${s}.sh"
	echo $jobfile
	cat <<EOF > $jobfile
#!/bin/bash
#SBATCH --nodes=1
#SBATCH --time=0-24:00:00
#SBATCH --mem=16G
#SBATCH --ntasks=4
#SBATCH -J trim
#SBATCH -o trim.%A.out
#SBATCH -e trim.%A.err

module load trim_galore
module load pigz
echo 'BEGIN'
date
hostname
trim_galore -j 2 --illumina --paired --fastqc -o /proj/marchlab/projects/PUPCYCLE_2019/PUP_SimUp/galore_test \\
$indir/${s}_R1_001.fastq.gz \\
$indir/${s}_R2_001.fastq.gz
echo 'END'
date

EOF

sbatch $jobfile

done 
```


